### PR TITLE
Include taxes migration in the states migrator helper method

### PR DESCRIPTION
**Pull request by Konamiman**
### All Submissions:

-   [ ] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

https://github.com/woocommerce/woocommerce/pull/35669 introduced a helper method to migrate state codes in various places of the database (shipping locations, orders, store address) but one placed that was missing is the tax rates table. This pull request fixes that.

### How to test the changes in this Pull Request:

Same testing instructions of https://github.com/woocommerce/woocommerce/pull/35669 but this time testing that the state codes in the tax rates (WooCommerce - Settings - Tax - Standard, Reduced and Zero taxes) are properly updated.

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/src/Database/Migrations/MigrationHelper.php [hunk 1]
```diff
@@ -262,4 +263,29 @@ private static function migrate_country_states_for_orders( string $country_code,
...
+	 * Migrate state codes in the tax rates table.
+	 *
+	 * @param string $country_code The country that has the states for which the migration is needed.
+	 * @param array  $old_to_new_states_mapping An associative array where keys are the old state codes and values are the new state codes.
+	 * @return void
+	 */
+	private static function migrate_country_states_for_tax_rates( string $country_code, array $old_to_new_states_mapping ): void {
+		global $wpdb;
+
+		// phpcs:disable WordPress.DB.PreparedSQL.NotPrepared
```
Comment (peterfabian) [inline plugins/woocommerce/src/Database/Migrations/MigrationHelper.php]: I think if you put the `$wpdb->prepare()` directly into the `$wpdb->query()`, the phpcs disabling won't be necessary. 

It's probably true for some other ones above, but this is fairly simple to change.

**Review by peterfabian — CHANGES_REQUESTED** [review]
Thanks, this works well. 

Besides the PHPCS below, I think we need to fix also the version strings in [class-wc-install](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-install.php#L220-L223)

```

'7.2.0' => array(
			'wc_update_720_adjust_new_zealand_states',
			'wc_update_720_adjust_ukraine_states',
		),

```

File: plugins/woocommerce/src/Database/Migrations/MigrationHelper.php (hunk 1 above)
Comment (Konamiman) [reply plugins/woocommerce/src/Database/Migrations/MigrationHelper.php]: I'll change it but in general I prefer to first generate the query in a separate variable and then run it, as the resulting code is cleaner than a nested `query(prepare($sql))`, even if it's at the expense of having to disable the check.

**Review by peterfabian — DISMISSED** [review]
Thanks, looks good to me now!

**Review by peterfabian — APPROVED** [review]
The failing e2e test is not related (p1671022234510819-slack-C03CPM3UXDJ), so approving after the PHPCS change again.

**Pull request merged**